WordPress/gutenberg
# i18n: Add pattern block

## Description

This PR is a proposal to add a pattern block. This will enable block theme developers to do a few things inside their templates:

- Output content containing strings that can be internationalized. 
- Output dynamic URLs see https://github.com/WordPress/twentytwentytwo/issues/92

For more context, see https://github.com/WordPress/gutenberg/pull/33192#issuecomment-874036632. This is another approach to that PR.

## How has this been tested?
- Check out this 2022 PR: https://github.com/WordPress/twentytwentytwo/pull/56
- Verify the credit in the footer appears as expected in the site editor and front-end
- Verify you can add the pattern in the editor
- Verify you can edit it in the site editor

## Screenshots 

https://user-images.githubusercontent.com/5375500/136443900-684a0450-ec02-494a-ba70-d5d407d321ec.mp4

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
